domain/domain.py

```diff
@@ -271,25 +288,34 @@ def get_domain_cell(self, attr, row):
 
         # Remove _nan_ if added due to correlated attributes
         domain.discard('_nan_')
+
         # Add initial value in domain
-        if pd.isnull(row[attr]):
-            domain.update(set(['_nan_']))
-            init_value = '_nan_'
-        else:
-            domain.update(set([row[attr]]))
-            init_value = row[attr]
-        return init_value, list(domain)
+        init_values = ['_nan_']
+        if not pd.isnull(row[attr]):
+            # Assume value in raw dataset is given as ||| separate initial values
+            init_values = row[attr].split('|||')
+        domain.update(set(init_values))
 
-    def get_random_domain(self, attr, cur_value):
+        # Take the first initial value as the current value
```

Comment:
Is that assignment of `current` going to change later on once we actually use `init_values`? If so, please add a TODO here

Reply:
We can definitely have a more principled way of this. Added a TODO.